### PR TITLE
fixing several odd cases

### PR DESCRIPTION
**Pull request by Galithil**
Direct dict access fails ont some projects, so I removed that to get a more coherent behaviour

Raising the error was handled by another try/except that has been removed, but it's not actually required. It would be better to not have the sample_run_metrics for the projects that lack the required info (usually, old projects)

### CODE REVIEW

**Comment by Galithil** [comment]
Also, someone is missing from the mail script.

File: LIMS2DB/objectsDB/objectsDB.py [hunk 1]
```diff
@@ -459,8 +459,7 @@ def _make_sample_run_id(self, seq_art, lims_run, prep, run_type):
                 samp_run_met_id = '_'.join([lane, date, fcid, barcode])
             except TypeError: 
                 #happens if the History object is missing fields, barcode might be None
-                logging.debug(self.name+" ",prep,"-", prep['reagent_label'])
-                raise TypeError
+                logging.debug("Missing field for making the sample run id :"+self.name+" ",prep,"-", prep['reagent_label'])
```
Comment (guillermo-carrasco) [inline LIMS2DB/objectsDB/objectsDB.py]: don't like `-format()` syntax?

Comment (Galithil) [inline LIMS2DB/objectsDB/objectsDB.py]: Good question .... why isnt that used ?

Comment (guillermo-carrasco) [inline LIMS2DB/objectsDB/objectsDB.py]: dunno man, but you fixed it :+1:

Comment (Galithil) [inline LIMS2DB/objectsDB/objectsDB.py]: There's still so much things in that code that don't make sense.

**Pull request merged**